File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-1.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        mp = {}
        left = 0
        ans = 0
        maxfreq = 0
        for right in range(len(s)):
            ch = s[right]
            if ch in mp:
                mp[ch] += 1
            else:
                mp[ch] = 1

            maxfreq = max(maxfreq , mp[ch])

            if (right-left+1) - maxfreq > k:
                mp[s[left]] -= 1
                left = left + 1

            ans = max(ans , right-left+1)

        return ans;

File: Data Structures & Algorithms/longest-palindromic-substring/submission-1.java
class Solution {
    public boolean isPalindrome(String sub){
        int left = 0;
        int right = sub.length()-1;
        while(left <= right){
            if(sub.charAt(left) != sub.charAt(right)){
                return false;
            }
            left++;
            right--;
        }
        return true;
    }
    public String longestPalindrome(String s) {
        String ans = "";
        for(int i=0;i<s.length();i++){
            for(int j=i;j<s.length();j++){
                String sub = s.substring(i,j+1);
                if(isPalindrome(sub) && sub.length() > ans.length()){
                    ans = sub;
                }
            }
        }
        return ans;
    }
        
}
